libs/dao/src/main/java/com/akto/dto/ApiCollectionUsers.java

```diff
@@ -56,6 +68,108 @@ public enum CollectionType {
             });
         }});
 
+    public static Set<String> getUrlsFromConditions(List<CollectionCondition> conditions){
+        Set<ApiInfoKey> apis = new HashSet<>();
+        conditions.forEach((condition) -> {
+            if(CollectionCondition.Operator.AND.equals(condition.getOperator())){
+                apis.retainAll(condition.returnApis());
+            } else {
+                apis.addAll(condition.returnApis());
+            }
+        });
+
+        Set<String> urls = new HashSet<>();
+        apis.forEach((api) -> {
+            urls.add(api.getUrl() + " " + api.getMethod());
+        });
+        return urls;
+    }
+
+    public static void updateApiCollection(List<CollectionCondition> conditions, int id) {
+        Set<String> urls = getUrlsFromConditions(conditions);
+
+        ApiCollectionsDao.instance.updateOne(
+                Filters.eq(Constants.ID, id),
+                Updates.combine(
+                    Updates.set(ApiCollection.CONDITIONS_STRING, conditions),
+                    Updates.set(ApiCollection.URLS_STRING, urls)));
+    }
+
+    private static Bson getFilters(List<CollectionCondition> conditions, CollectionType type){
+        List<Bson> filters = new ArrayList<>();
+        List<Bson> orFilters = new ArrayList<>();
+        conditions.forEach((condition) -> {
+            Bson conditionFilter = condition.returnFiltersMap().get(type);
+            if(condition.getOperator().equals(CollectionCondition.Operator.OR)){
+                orFilters.add(conditionFilter);
+            } else {
+                filters.add(conditionFilter);
+            }
+        });
+        if(!orFilters.isEmpty()){
+            filters.add(Filters.or(orFilters));
+        }
+        return Filters.and(filters);
+    }
+
+    public static void operationForCollectionId(List<CollectionCondition> conditions, int apiCollectionId, Bson update, Bson matchFilter, boolean remove) {
+        Map<CollectionType, Bson> filtersMap = new HashMap<>();
+        for (CollectionType type : CollectionType.values()) {
+            Bson filter = getFilters(conditions, type);
+            if(remove){
+                filter = Filters.nor(filter);
+            }
+            Bson finalFilter = Filters.and(matchFilter, filter);
+            filtersMap.put(type, finalFilter);
+        }
+        updateCollectionsForCollectionId(filtersMap, update);
+    }
+
+    public static void addToCollectionsForCollectionId(List<CollectionCondition> conditions, int apiCollectionId) {
+        Bson update = Updates.addToSet(SingleTypeInfo._COLLECTION_IDS, apiCollectionId);
+        Bson matchFilter = Filters.nin(SingleTypeInfo._COLLECTION_IDS, apiCollectionId);
+        operationForCollectionId(conditions, apiCollectionId, update, matchFilter, false);
+    }
+
+    public static void removeFromCollectionsForCollectionId(List<CollectionCondition> conditions, int apiCollectionId) {
```

Comment:
Can't follow much of this. These are all `public` methods?

Reply:
Yes, apart from one or two methods, most are called outside.